DashWallet/Sources/UI/Coinbase/Custodial Swaps/Model/CustodialSwapsModel.swift

```diff
@@ -0,0 +1,159 @@
+
+    override var isSendAllowed: Bool {
+        selectedAccount != nil && amount.plainAmount > 0 && !canShowInsufficientFunds
+    }
+
+    override var canShowInsufficientFunds: Bool {
+        guard let selectedAccount else { return false }
+
+        let plainAmount = amount.plainAmount
+        return plainAmount > selectedAccount.info.plainAmountInDash
+    }
+
+    override init() {
+        super.init()
+    }
+
+    func convert() {
+        guard let selectedAccount, let dashAccount = Coinbase.shared.dashAccount else { return }
+
+        guard let usdAmount = try? Coinbase.shared.currencyExchanger.convertDash(amount: amount.plainAmount.dashAmount, to: "USD"),
+              usdAmount > 2 else {
+            let min = NSDecimalNumber(decimal: kMinUSDAmountOrder)
+            let localFormatter = NumberFormatter.fiatFormatter(currencyCode: "USD")
+            let str = localFormatter.string(from: min) ?? "$1.99"
+
+            error = Coinbase.Error.transactionFailed(.enteredAmountTooLow(minimumAmount: str))
+            return
+        }
+
+        Task {
+            do {
+                guard let originAmount = try? Coinbase.shared.currencyExchanger.convertDash(amount: amount.plainAmount.dashAmount, to: selectedAccount.info.currencyCode) else {
+                    return
+                }
+
+                let result = try await Coinbase.shared.placeTradeOrder(from: selectedAccount,
+                                                                       to: dashAccount,
+                                                                       amount: originAmount.string)
+
+                await MainActor.run { [weak self] in
+                    self?.delegate?.custodialSwapsModelDidPlace(order: result)
+                }
+            } catch {
+                await MainActor.run { [weak self] in
+                    self?.error = error
+                }
+            }
+        }
+    }
+
+    override func selectAllFundsWithoutAuth() {
+        guard let selectedAccount else { return }
+
+        let max = AmountObject(localAmountString: selectedAccount.info.balance.amount, fiatCurrencyCode: selectedAccount.info.balance.currency,
+                               localFormatter: supplementaryNumberFormatter)
+        supplementaryAmount = max
+        mainAmount = supplementaryAmount.dashAmount
+        amountDidChange()
+    }
+
+    override func checkAmountForErrors() {
+//        guard let usdAmount = try? Coinbase.shared.currencyExchanger.convertDash(amount: amount.plainAmount.dashAmount, to: "USD"),
+//                usdAmount > 2 else {
+//            let min = NSDecimalNumber(decimal: kMinUSDAmountOrder)
+//            let localFormatter = NumberFormatter.fiatFormatter(currencyCode: "USD")
+//            let str = localFormatter.string(from: min) ?? "$1.99"
+//
+//            error = Coinbase.Error.transactionFailed(.enteredAmountTooLow(minimumAmount: str))
+//            return
+//        }
```

Comment:
Not needed?

Reply:
@Syn-McJ Thanks, I've removed commented code.